pymem/pattern.py

```diff
@@ -64,18 +66,23 @@ def scan_pattern_page(handle, address, pattern, *, return_multiple=False):
 
     page_bytes = pymem.memory.read_bytes(handle, address, mbi.RegionSize)
 
+    if scan_hex:
+        page_bytes = page_bytes.hex().encode()
+        pattern = pattern.lower()
```

Comment:
Pattern is made lowercase since the actual hex values are lowercase. Removes a potential user issue where they might write their hex out using capital letters